Small edit to prevent the last saved query/mutation from executing ...

jthorn opened the pull request:
when the add-in is loaded.

I wanted to change this because I was debugging a mutation, and when I loaded the add-in again later, The previous mutation automatically executed. This behavior can have some pretty serious consequences if the mutation previously run modifies any significant amount of data, or kicks off some long running process.

The behavior of executing the last query or mutation on change of endpoint is unaffected.

jthorn committed:
Small edit to prevent the last saved query/mutation from executing when the add-in in loaded.

PowerKiKi commented:
Thanks for the contribution. I have had similar experiences. I wonder if we should disable auto-executing entirely. I can easily imagine dangerous cases when changing endpoints too (eg: between staging and production). WDYT ?

jthorn commented:
I thought about that too. But auto-execute seems reasonable in some cases. I often execute the same queries against multiple endpoints. But yeah, staging vs. prod is an excellent example.

Also, I didn't want to make things too complicated for this release. just simply address the issue at hand.

My thought for another edit would be to add a checkbox next to the button that allows a user to specify the default behavior they want. Checked = Auto-execute ... Unchecked = No Execute.

It really just depends on how complicated you want to make the add-in if the stated goal is the thinnest possible wrapper :)

If adding options is more bloat than it's worth, then yes, I agree it would be better to remove it altogether.

PowerKiKi commented:
Let's remove it altogether then.

Would you be willing to update your PR, to remove auto-executing, and to clear the result box when endpoints change ? So that we don't show an outdated result for a new endpoint ?

jthorn commented:
I can remove the auto-execute pretty easily. Clearing the results on change will take me a little bit of research since the results are a part of the GraphiQL interface. I can tinker with that later. There are also some idiosyncrasies with the state management that I would like to look into, like the change endpoint button not recognizing the value of the textbox on initial load (although it's a narrow edge case).

Give me a minute to test my removal, and I'll update the pull request.

jthorn committed:
Removed logic to auto-execute queries on change of endpoint.

jthorn commented:
Ok. I removed the block altogether. There are still some "oddness" I wasn't able to track down when changing endpoints. It appears that the GraphiQL component attempts to connect with the new endpoint and validate the current query, and if the query is not valid for the endpoint, it throws a validation error. But it seems to be benign.

I validated that when an endpoint is changed, and a query that is valid for both endpoints is current, it does not actually execute the query on either load, of click of the change endpoint button.

jthorn committed:
Updated state management and refactored code as a function component using react hooks

jthorn commented:
Sorry if that last commit it too much. I'm used to writing React in typescript using hooks. If you want to reject the edit because of the drastic nature of the refactor, I understand. I initially set out only to make a minor change.

That said, the URL parsing and parameter passing to the GraphiQL component seemed superfluous. I ended up removing it and the resulting code is much smaller and easier to follow (IMHO)

PowerKiKi committed:
Drop endpoint button

Because queries are not replayed automatically when endpoint changes,
the button actually has no effect at all. And it could be confusing
to see an endpoint in input that was not yet applied via the button
with a mis-matching query results.

Removing the button entirely, avoid confusing states.

PowerKiKi commented:
I simplified things even further, even removing the button altogether, since it does not have any real use anymore.

As a sidenote `graphQLFetcher` is back outside of the `GraphiQLExtension` to avoid re-creating the exact same function on every rendering.

Most importantly inspection queries are not sent on every keystroke anymore.

Would you have any other PR in mind ? or shall we release this ?

jthorn commented:
Looks good to me ... ship it!

:)

PowerKiKi commented:
:rocket: pending review on the store
